jobs/build-kola-containers.Jenkinsfile

```
@@ -0,0 +1,209 @@
+    def output = shwrapCapture("git rev-parse HEAD")
+    commit = output.substring(0,40)
+    shortcommit = commit.substring(0,7)
+
+    def path=params.PATH_TO_CONTEXTS
+    // Check for changes in tests/containers/*
+    // If not, no need to run this
+    // not using shWrapCapture here because i don't want -euxo pipefail to be set, as grep returns 1 on non-match
+    def changeset = sh(returnStatus: true, script:"git diff-tree -r --name-only HEAD | grep ${path}")
+    if ( changeset == 1 && ! params.FORCE ) {
+        changes = false
+    }
+
+    // gather the context folders list
+    contexts = shwrapCapture("""
+        cd ${path}
+        find . -maxdepth 1 -mindepth 1 -type d -exec basename {} \\;
+    """).trim().split("\n")
+}
+
+if ( !changes ) {
+    currentBuild.result = 'SUCCESS'
+    currentBuild.description = "[${gitref}@${shortcommit}] No changes in ${path}. Skiped the build."
+    return
+}
+
+currentBuild.description = "[${gitref}@${shortcommit}] Waiting"
+
+// Get the list of requested architectures to build for
+def basearches = params.ARCHES.split() as Set
+// and  the list of images to build
+def imageNames = contexts as Set
+
+lock(resource: "build-containers") {
+    cosaPod(image: "quay.io/coreos-assembler/coreos-assembler",
+            memory: "512Mi", kvm: false,
+            serviceAccount: "jenkins") {
+    timeout(time: 60, unit: 'MINUTES') {
+    try {
+
+        currentBuild.description = "[${gitref}@${shortcommit}] Running"
+
+        // By default we will allow re-using cache layers for one day.
+        // This is mostly so we can prevent re-downloading the RPMS
+        // and repo metadata and over again in a given day for successive
+        // builds.
+        def cacheTTL = "24h"
+        def force = ""
+        if (params.FORCE) {
+            force = '--force'
+            // Also set cacheTTL to 0.1s to allow users an escape hatch
+            // to force no cache layer usage.
+            cacheTTL = "0.1s"
+        }
+
+        withCredentials([file(credentialsId: 'cosa-push-registry-secret', variable: 'REGISTRY_SECRET')]) {
+            stage('Build Containers') {
+                parallel basearches.collectEntries{arch -> [arch, {
+                    for (imageName in imageNames) {
+                        def dirname = params.PATH_TO_CONTEXTS + imageName
+                        pipeutils.withPodmanRemoteArchBuilder(arch: arch) {
+                            shwrap("""
+                                cosa remote-build-container \
+                                    --git-sub-dir $dirname\
+                                    --arch $arch --cache-ttl ${cacheTTL} \
+                                    --git-ref $commit ${force} \
+                                    --git-url ${params.COREOS_ASSEMBLER_GIT_URL} \
+                                    // Force the tag to inclue the image name to avoid racing between aarches
+                                    // and overwrite the images
```

```suggestion
                                    // Force the tag to include the image name to avoid racing between arches
                                    // and overwrite the images
```

That's neat. I guess those lines correctly get parsed as Groovy comments because it happens before the parsing of the multi-line string. My brain keeps reading it as invalid.

---

No you are right, it does not work :) Fixed